src/coreclr/jit/importer_vectorization.cpp

```diff
@@ -199,22 +286,61 @@ GenTree* Compiler::impExpandHalfConstEqualsSIMD(GenTreeLclVar* data, WCHAR* cns,
 //  |     \--*  CNS_INT   <offset>
 //  \--*  CNS_INT   <value>
 //
+//  or in case of 'ignoreCase':
+//
+//  *  EQ        int
+//  +--*  OR        int
+//  |  +--*  IND       <type>
+//  |  |  \--*  ADD       byref
+//  |  |     +--*  <obj>
+//  |  |     \--*  CNS_INT   <offset>
+//  |  \--*  CNS_INT   <lowercase mask>
+//  \--*  CNS_INT   <lowercased value>
+//
 // Arguments:
-//    comp -       Compiler object
-//    obj -        GenTree representing data pointer
-//    type -       type for the IND node
-//    offset -     offset for the data pointer
-//    value -      constant value to compare against
+//    comp       - Compiler object
+//    obj        - GenTree representing data pointer
+//    type       - Type for the IND node
+//    offset     - Offset for the data pointer
+//    value      - Constant value to compare against
+//    ignoreCase - Ignore case mode (works only for ASCII cns)
+//    useXor     - Use GT_XOR instead of GT_EQ
 //
 // Return Value:
 //    A tree with indirect load and comparison
-//
-static GenTree* impCreateCompareInd(Compiler* comp, GenTreeLclVar* obj, var_types type, ssize_t offset, ssize_t value)
+//    nullptr in case of 'ignoreCase' mode and non-ASCII value
+//
+static GenTree* impCreateCompareInd(Compiler*      comp,
+                                    GenTreeLclVar* obj,
+                                    var_types      type,
+                                    ssize_t        offset,
+                                    ssize_t        value,
+                                    bool           ignoreCase,
+                                    bool           useXor = false)
```

Comment:
Consider capturing the various options as an enum -- it's more readable at the call sites. Though sadly with C++, it's a bit clunky to have flags-style enums.

Reply:
Done